tools/rofiTool/build.cpp

```diff
@@ -1,88 +1,127 @@
-        case BotType::UMpadN:
-            n = readIntWithMsg( "Dimension: " );
+        }
+        case BotType::UMpadN: {
+            int n = readIntWithMsg( "Dimension: " );
             return buildUMpad( n );
-        case BotType::UMpadNM:
-            n = readIntWithMsg( "Dimension n: " );
-            m = readIntWithMsg( "Dimension m: " );
+        }
+        case BotType::UMpadNM: {
+            int n = readIntWithMsg( "Dimension n: " );
+            int m = readIntWithMsg( "Dimension m: " );
             return buildUMpad( n, m );
-        case BotType::NPad:
-            n = readIntWithMsg( "Dimension: " );
+        }
+        case BotType::NPad: {
+            RofiWorld world;
+            int n = readIntWithMsg( "Dimension: " );
             world.insert( Pad( 0, n ) );
-            break;
-        case BotType::NMPad:
-            n = readIntWithMsg( "Dimension n: " );
-            m = readIntWithMsg( "Dimension m: " );
+            return world;
+        }
+        case BotType::NMPad: {
+            RofiWorld world;
+            int n = readIntWithMsg( "Dimension n: " );
+            int m = readIntWithMsg( "Dimension m: " );
             world.insert( Pad( 0, n, m ) );
-            break;
-        default:
-            throw std::runtime_error( "Unknown model" );
+            return world;
+        }
     }
-    return world;
+
+    ROFI_UNREACHABLE( "Unknown BotType" );
 }
 
-std::string botTypeToString( BotType b ) {
-    std::string str;
+std::string botTypeToString( BotType b )
+{
     switch ( b ) {
         case BotType::UM:
-            str = "an universal module";
-            break;
+            return "an universal module";
         case BotType::UMpadN:
         case BotType::UMpadNM:
-            str = "a pad of universale modules";
-            break;
-        default:
-            str = "a RoFICoM pad";
+            return "a pad of universale modules";
+        case BotType::NPad:
+        case BotType::NMPad:
+            return "a RoFICoM pad";
     }
-    return str;
+
+    ROFI_UNREACHABLE( "Unknown BotType" );
 }
 
-int build( Dim::Cli & /* cli */ ) {
-    auto world = buildWishedRofibot( static_cast< BotType >( *botType ) );
+void build( Dim::Cli & cli )
+{
+    using namespace rofi::configuration;
+
+    auto bType = toBotType( *botType );
+    if ( !bType ) {
+        cli.fail( EXIT_FAILURE, "Invalid value for BotType" );
+        return;
+    }
 
-    rofi::configuration::connect< rofi::configuration::RigidJoint >(
-        *botType > 2 ? world.getModule( 0 )->connectors()[ 0 ] // No body within the pad
-                     : world.getModule( 0 )->bodies()[ 0 ],     // UM, so we fix its body
-        rofi::configuration::matrices::Vector( { 0, 0, 0 } ),
-        rofi::configuration::matrices::identity );
+    auto world = buildWishedRofibot( *bType );
+    if ( world.modules().empty() ) {
+        cli.fail( EXIT_FAILURE, "Empty rofi world" );
+        return;
+    }
+    affixRofiWorld( world );
```

Comment:
To be consistent with my other comment to these lines: here adding a RigidJoint using `affixRofiWorld `is required, since `buildWishedRofibot `does not add any on its own (I think), and the following code assumes the world can be prepared (therefore it is affixed in space in at least one point).